src/common/util/mfe.js

```diff
@@ -24,14 +24,17 @@ export function hasValidValue (val) {
  * @returns {{'mfe.id': *, 'mfe.name': String}|{}} returns an empty object if args are not supplied or the aggregate instance is not supporting version 2
  */
 export function getVersion2Attributes (target, aggregateInstance) {
-  if (aggregateInstance?.harvestEndpointVersion !== 2) return {}
-  const containerAgentEntityGuid = aggregateInstance.agentRef.runtime.appMetadata.agents[0].entityGuid
-  if (!isValidMFETarget(target)) {
+  /** If the feature doesnt support registered entities (V2), no need to add any attributes at all */
+  if (!aggregateInstance.supportsRegisteredEntities) return {}
+  const containerAgentEntityGuid = aggregateInstance?.agentRef.runtime.appMetadata.agents[0].entityGuid
+  /** If the target in question is not an MFE (container), or its "blocked" (feature flag or register issue), just report it to the container */
+  if (!isValidMFETarget(target) || target.blocked) {
```

Comment:
Note that this block doesn't match the function description. Rather, this check belongs in whatever calls "get me version 2 attributes", and brings confusion by sometimes returning version 1 attributes instead.

Reply:
These are the version 2 attributes for an invalid target tho... (ie, the container agent wont have a target, but if it still needs to report to v2 it needs those attributes as body attributes)